### PR TITLE
Commands: save

### PR DESCRIPTION
**Pull request by leezahuor**
# Issue/Task:
https://fridgeventory.atlassian.net/browse/FRID-35
This is implementing the `save()` method in Commands. 

This method should: 

- Prompt users to create or load a profile if not already done.
- Allow users to save their profile information as a JSON. 

Additional changes included:

- Adding `asDictionary()` methods to foodStorage and Food.
- Adding `save()` method to Profile. 
- Updating commands_messages and constants. 

# Checklist
- [X] Fixed Issue/Accomplished Task
- [X] Wrote tests
- [X] All test cases passing

### CODE REVIEW

File: src/Constants/constants.py [hunk 1]
```diff
@@ -8,7 +8,11 @@
 DELETE_FREEZER = "delete freezer"
 DELETE_FRIDGE = "delete fridge"
 DELETE_PROFILE = "delete profile"
+EXPIRATION_DATE = "expiration date"
+EXPIRATION_WINDOW = "expiration window"
```
Comment (amandate) [inline src/Constants/constants.py:12]: These should both have underscores instead of spaces

File: src/Constants/constants.py [hunk 2]
```diff
@@ -27,7 +32,9 @@
 PROFILES_PATH = "profiles"
 REMOVE_FOOD = "remove food"
 SAVE = "save"
+SLASH = "/"
 SPACE = " "
 TAB = "    "
+USE_BY_DATE = "use by date"
```
Comment (amandate) [inline src/Constants/constants.py:38]: Same comment as above

File: src/Session/commands.py [hunk 3]
```diff
@@ -127,8 +127,16 @@ def open(self, foodStorage_type):
         # If food storage exists with this name, prints open success message. 
         print(OPEN_FOOD_STORAGE_SUCCESS_MESSAGE.format(foodStorage_type, foodStorage_name))
                    
+    ''' Allows user to save their profile information. '''
```
Comment (amandate) [inline src/Session/commands.py:130]: Should include that it saves it to a json file under the profiles folder

File: src/Session/profile.py [hunk 4]
```diff
@@ -59,5 +61,26 @@ def listFoodStorages(self, foodStorage_type):
...
 
+    ''' Takes information provided by user and creates a json with the profile information saved to it. '''
     def save(self):
-        pass
\ No newline at end of file
+        user_profile = {
+            NAME : self.name, 
+            FOOD_STORAGES : {
+                
+            }
```
Comment (amandate) [inline src/Session/profile.py:70]: this can be on one line

File: tests/FoodStorage/test_foodStorage.py [hunk 5]
```diff
@@ -108,6 +109,14 @@ def testUpdateFoods(self):
...
         self.foodStorage.update(name1)
         self.assertEqual([food2, food1], self.foodStorage.list())
+    
+    def testAsDictionary(self):
+        actual = self.foodStorage.asDictionary() 
+        expected = {
+            EXPIRATION_WINDOW : 7,
+            FOOD : [],
+        } 
+        self.assertEqual(expected, actual)
```
Comment (amandate) [inline tests/FoodStorage/test_foodStorage.py:120]: I think you should test the scenario for if there's food in the foodstorage

File: tests/Session/test_commands.py [hunk 6]
```diff
@@ -233,5 +234,41 @@ def testOpenFoodStorage(self, mocked_input):
...
+        self.commands.save()
+
+        expectedOutput = SAVE_PROFILE_SUCCESS_MESSAGE + NEW_LINE
+        self.assertEqual(expectedOutput, capturedPrintOutput.getvalue())
+
+        # reset standout
+        sys.stdout = sys.__stdout__
+
+        ## May add another test with food added after merge w/ add_foods ##
+
```
Comment (amandate) [inline tests/Session/test_commands.py:272]: this isn't necessary in this case

File: tests/Session/test_profile.py [hunk 7]
```diff
@@ -47,6 +49,64 @@ def testListFoodStorages(self):
...
+        profile3.addFoodStorage(FREEZER, freezer_name)
+        profile3.save()
+        with open(TEST_PROFILES_PATH + PROFILE_W_MULTI_FOOD_STORAGE + JSON_EXTENSION) as user_file:
+            testProfile = json.load(user_file)
+        with open(PROFILES_PATH + SLASH + TEST_PROFILE_NAME_3 + JSON_EXTENSION) as user_profile:
+            savedProfile = json.load(user_profile)
+
+        self.assertEqual(testProfile, savedProfile)
+
+        # ## will add after merge with add_food command ##
```
Comment (amandate) [inline tests/Session/test_profile.py:95]: your tests here look good so far! just waiting for the last test when add_food gets merged!

**Review by amandate — COMMENTED** [review]
mostly small things but looks good so far!

File: src/Constants/constants.py (hunk 1 above)
Comment (leezahuor) [reply src/Constants/constants.py:12]: Done!

File: src/Constants/constants.py (hunk 2 above)
Comment (leezahuor) [reply src/Constants/constants.py:38]: Done!

File: src/Session/commands.py (hunk 3 above)
Comment (leezahuor) [reply src/Session/commands.py:130]: Done!

File: src/Session/profile.py (hunk 4 above)
Comment (leezahuor) [reply src/Session/profile.py:70]: Got it!

File: tests/Session/test_profile.py (hunk 7 above)
Comment (leezahuor) [reply tests/Session/test_profile.py:95]: Thank you!

File: tests/Session/test_commands.py (hunk 6 above)
Comment (leezahuor) [reply tests/Session/test_commands.py:272]: Got it!

File: tests/FoodStorage/test_foodStorage.py (hunk 5 above)
Comment (leezahuor) [reply tests/FoodStorage/test_foodStorage.py:120]: Done, let me know what you think!

**Review by amandate — COMMENTED** [review]
LGTM so far! Just waiting for that last test when add_foods gets merged :) Good job!!

**Review by amandate — APPROVED** [review]
Looks solid!

**Pull request merged**